### PR TITLE
if possible, harmonize non semver compatible image versions

### LINKED ISSUE #1467: Improve problem visualization for clusters with machine image with an invalid semver version
**What would you like to be added**:
If there is a cluster with a machine image taht has an invalid semver version e.g. `1.2` or `1.2.3.4` the image will be droped in the list of valid machine images. On the cluster details page an error is shown taht the machine image is expired. The error should be that the machine image or better the semver version of the machine image is invalid.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by grolu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1467

**Special notes for your reviewer**:

**Release note**:

```feature user
Added Fallback Solution for Non-SemVer Compatible Machine Image Versions: In previous versions, machine images from some vendors that did not adhere to Semantic Versioning (SemVer) standards were omitted from the user view in the dashboard. This was due to the system's strict adherence to SemVer for sorting and presenting image versions, resulting in non-compliant versions being filtered out. To address this, we now try to harmonize those versions to make them SemVer compliant. If this is not possible, affected versions will still be filtered out.
```

### CODE REVIEW

File: frontend/src/store/cloudProfile/index.js [hunk 1]
```diff
@@ -374,13 +375,18 @@ export const useCloudProfileStore = defineStore('cloudProfile', () => {
...
+            versionObj.version = harmonizedVersion
+          } else {
+            logger.error(`Skipped machine image ${machineImage.name} as version ${versionObj.version} is not a valid semver version and cannot be harmonized`)
+            return undefined
+          }
         }
-        return true
-      })
+        return versionObj
+      }))
```
Comment (petersutter) [inline frontend/src/store/cloudProfile/index.js:389]:
```js
      const versions = compact(map(machineImage.versions, versionObj => {
        if (semver.valid(versionObj.version)) {
          return versionObj
        }

        const harmonizedVersion = harmonizeVersion(versionObj.version)
        if (!harmonizedVersion) {
          logger.error(`Skipped machine image ${machineImage.name} as version ${versionObj.version} is not a valid semver version and cannot be harmonized`)
          return undefined
        }

        versionObj.version = harmonizedVersion

        return versionObj
      }))
```

File: frontend/src/utils/index.js [hunk 2]
```diff
@@ -655,3 +660,11 @@ export function parseNumberWithMagnitudeSuffix (abbreviatedNumber) {
   const factor = suffixFactors[suffix?.toLowerCase()] ?? 1
   return Number(number) * factor
 }
+
+export function harmonizeVersion (version) {
```
Comment (holgerkoser) [inline frontend/src/utils/index.js:664]: rename to `normalizeVersion`

File: frontend/src/utils/index.js [hunk 3]
```diff
@@ -44,6 +44,12 @@ import {
 } from '@/lodash'
 
 const serviceAccountRegex = /^system:serviceaccount:([^:]+):([^:]+)$/
+const sizeRegex = /^(\d+)Gi$/
+const emailRegex = /^[a-zA-Z0-9.!#$%&’*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/
+const colorCodeRegex = /^#([a-f0-9]{6}|[a-f0-9]{3})$/i
+const magnitudeNumberSuffixRegex = /^(\d+(?:\.\d*)?)([kmbt]?)$/i
+const versionRegex = /^(\d+)(?:\.(\d+))?(?:\.(\d+))?(?:\.\d+)*(([-+]).+)?$/
```
Comment (holgerkoser) [inline frontend/src/utils/index.js:51]:
```suggestion
const versionRegex = /^(\d+)(?:\.(\d+))?(?:\.(\d+))?(?:\.\d+)*([-+].+)?$/
```

File: frontend/src/utils/index.js [hunk 4]
```diff
@@ -655,3 +660,11 @@ export function parseNumberWithMagnitudeSuffix (abbreviatedNumber) {
   const factor = suffixFactors[suffix?.toLowerCase()] ?? 1
   return Number(number) * factor
 }
+
+export function harmonizeVersion (version) {
+  const [match, seg1, seg2, seg3, suffix] = versionRegex.exec(version) ?? []
+  if (match) {
+    return `${Number(seg1)}.${Number(seg2 ?? 0)}.${Number(seg3 ?? 0)}${suffix ?? ''}`
+  }
+  return undefined
```
Comment (holgerkoser) [inline frontend/src/utils/index.js:669]:
```suggestion
  const [match, major, minor = '0', patch = '0', suffix = ''] = versionRegex.exec(version) ?? []
  if (match) {
    return [major, minor, patch].join('.') + suffix
  }
```

File: frontend/src/store/cloudProfile/index.js (hunk 1 above)
Comment (holgerkoser) [reply frontend/src/store/cloudProfile/index.js:389]:
```js
const versions = []
for (const versionObj of machineImage.versions) {
  if (semver.valid(versionObj.version)) {
    versions.push(versionObj)
    continue
  }

  const normalizedVersion = normalizeVersion(versionObj.version)
  if (normalizedVersion) {
    versionObj.version = normalizedVersion
    versions.push(versionObj)
    continue
  }

  logger.error(`Skipped machine image ${machineImage.name} as version ${versionObj.version} is not a valid semver version and cannot be harmonized`)
}
```

**Pull request merged**